libraries/AP_NavEKF3/AP_NavEKF3_core.h

```diff
@@ -1323,7 +1322,7 @@ class NavEKF3_core : public NavEKF_core_common
         float innovVar;     // innovation variance (m^2)
         float testRatio;    // innovation consistency test ratio
         Vector3f beaconPosNED; // beacon NED position
-    } rngBcnFusionReport[10];
+    } rngBcnFusionReport[4];
```

Comment:
What happens if beacon->count(); returns a value large than 4?

Reply:
I see that AP_BEACON_MAX_BEACONS is currently set to 4, but that is in a different library, so adding an upper limit on beacon->count() in NavEKF3_core::readRngBcnData() would be advisable.

Reply:
yes, it will go badly.. just like if it had returned a number > 10 before.  I'll add a check, thanks!